koku/api/tests_query_handler.py

```diff
@@ -0,0 +1,125 @@
+                                                                                            'value': Decimal('2186.508029214')}, # noqa
+                                                                    'instance_type': 'r5.2xlarge',
+                                                                    'markup_cost': {'units': 'USD',
+                                                                                     'value': Decimal('218.650802921')},
+                                                                    'region': 'eu-west-3',
+                                                                    'service': 'AmazonEC2',
+                                                                    'usage': {'units': 'Hrs',
+                                                                                'value': Decimal('3807.000000000')}}]}],
+                                    'service': 'AmazonEC2'}]}]},
+        {'date': '2019-12-02', 'regions': []},
+        {'date': '2019-12-03', 'regions': []},
+        {'date': '2019-12-04', 'regions': []},
+        {'date': '2019-12-05', 'regions': []},
+        {'date': '2019-12-06', 'regions': []},
+        {'date': '2019-12-07', 'regions': []},
+        {'date': '2019-12-08', 'regions': []},
+        {'date': '2019-12-09', 'regions': []}],
+
+        """
+        self.generator.add_data_to_tenant(FakeAWSCostData(self.provider), product='ec2')
+        self.generator.add_data_to_tenant(FakeAWSCostData(self.provider, region='eu-west-3'), product='ec2')
+        self.generator.add_data_to_tenant(FakeAWSCostData(self.provider, region='us-west-1'), product='ec2')
+
+        # First Request:
+        url = '?group_by[region]=*&filter[region]=eu-west-3&group_by[service]=AmazonEC2'
+        query_params = self.mocked_query_params(url, AWSInstanceTypeView)
+        handler = AWSReportQueryHandler(query_params)
+        data = handler.execute_query()
+        # Second Request:
+        url2 = '?group_by[region]=eu-west-3&group_by[service]=AmazonEC2'
+        query_params2 = self.mocked_query_params(url2, AWSInstanceTypeView)
+        handler2 = AWSReportQueryHandler(query_params2)
+        data2 = handler2.execute_query()
+        # Assert the second request contains only eu-west-3 region
+        for region_dict in data2['data']:
+            # For each date, assert that the region is eu-west-3
+            for list_item in region_dict['regions']:
+                self.assertEquals('eu-west-3', list_item['region'])
+        # Assert the first request contains only eu-west-3
+        for region_dict in data['data']:
+            # For each date, assert that the region is eu-west-3
+            for list_item in region_dict['regions']:
+                self.assertEquals('eu-west-3', list_item['region'])
+
+    def test_filter_to_group_by(self):
+        """Test the filter_to_group_by method."""
+        url = '?group_by[region]=*&filter[region]=eu-west-3&group_by[service]=AmazonEC2'
```

Comment:
You should add a test case whit multiple filters of the same value.
`url=?group_by[region]=*&filter[region]=eu-west-3&filter[region]=us-east-1`

You should add a test case without group_by in the query parameters:
`url=?filter[region]=eu-west-3`

I believe these two tests would uncover some of the comments above related to dictionary access and the filter list.

Reply:
Will do

Reply:
What is the desired output in the case of two filters?

Reply:
Would this query be an equivalent replacement for multiple filters?
`group_by[service]=eu-west-3&group_by[service]=us-east-1`

Reply:
Yes. 

And the second is a sanity check related to the group_by access.  Should run with out error and essentially leaving the parameters untouched since it should skip your group_by cases.

Reply:
Added